model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,643 @@
+      0x9 => {stack_adj_base = zero_extend(0x20);},
+      0xa => {stack_adj_base = zero_extend(0x20);},
+      0xb => {stack_adj_base = zero_extend(0x20);}, 
+      0xc => {stack_adj_base = zero_extend(0x30);}, 
+      0xd => {stack_adj_base = zero_extend(0x30);}, 
+      0xe => {stack_adj_base = zero_extend(0x30);}, 
+      0xf => {stack_adj_base = zero_extend(0x40);}, 
+    };
+    } else {
+    match rlist {
+      0x4 => {stack_adj_base = zero_extend(0x10);},
+      0x5 => {stack_adj_base = zero_extend(0x10);},
+      0x6 => {stack_adj_base = zero_extend(0x20);}, 
+      0x7 => {stack_adj_base = zero_extend(0x20);},
+      0x8 => {stack_adj_base = zero_extend(0x30);},
+      0x9 => {stack_adj_base = zero_extend(0x30);}, 
+      0xa => {stack_adj_base = zero_extend(0x40);},
+      0xb => {stack_adj_base = zero_extend(0x40);}, 
+      0xc => {stack_adj_base = zero_extend(0x50);}, 
+      0xd => {stack_adj_base = zero_extend(0x50);}, 
+      0xe => {stack_adj_base = zero_extend(0x60);}, 
+      0xf => {stack_adj_base = zero_extend(0x70);},
+    };
+  };
+
+  let stack_adj_by : bits(xlen) = zero_extend(sp54 << 4);
+  stack_adj = stack_adj_base + stack_adj_by;
+
+  var addr : bits(xlen) = X(sp) + stack_adj - zero_extend(bytes);
+
+  foreach (i from 0 to 12) {
+    //if register i is in xreg_list
+    if (reg_list[i] != 0b00000) then {
+      if (bytes == 0x4) then {
+        let rs1 = reg_list[i];
+        let imm : bits(xlen) = stack_adj - zero_extend(bytes);
+        let _ = execute(LOAD(imm[11..0], sp, rs1, false, WORD, false, false))
+      } else {
+        let rs1 = reg_list[i];
+        let imm : bits(xlen) = stack_adj - zero_extend(bytes);
+        let _ = execute(LOAD(imm[11..0], sp, rs1, false, DOUBLE, false, false))
+      };
+      addr = addr - zero_extend(bytes);
+    }
+  };
+  let _ = execute(C_LI(0b000000, 0b01010));
+  X(sp) = X(sp) + stack_adj;
+  let _ = execute(RISCV_JALR(0b000000000000, ra, zreg));
+
+
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = CM_POPRETZ(rlist, sp54) if (sizeof(xlen) == 32) <->
+  "cm.popretz" ^ spc() ^ CM_POP_MAPPING_32(rlist, sp54) if (sizeof(xlen) == 32)
+
+mapping clause assembly = CM_POPRETZ(rlist, sp54) if (sizeof(xlen) == 64) <->
+  "cm.popretz" ^ spc() ^ CM_POP_MAPPING_64(rlist, sp54) if (sizeof(xlen) == 64)
+
+
+// cm.popret
+union clause ast = CM_POPRET : (bits(4), bits(2))
+
+mapping clause encdec_compressed = CM_POPRET(rlist, sp54)                if extensionEnabled(Ext_Zcmp)
+  <-> 0b101 @ 0b11110 @ rlist : bits(4) @ sp54 : bits(2) @ 0b10        if extensionEnabled(Ext_Zcmp)
+
+function clause execute (CM_POPRET(rlist, sp54)) = {
+
+  var bytes : bits(4) = zeros();
+
+  if (sizeof(xlen) == 32) then {
+    bytes = 0x4
+  };
+
+  if (sizeof(xlen) == 64) then {
+    bytes = 0x8
+  };
+
+  reg_list : vector (13, dec, bits(5)) = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000];
+
+  match rlist {
```

Comment:
This rlist seems duplicated between functions. If so, I think it should be factored into a helper function. I think it also needs some kind of comment, i.e. why do we jump straight from `0b00001` to `0b01000`?

Reply:
(Note that this is more of a rhetorical question, as I know it's because x8 is the first saved register and x1 is the return address).